libsolidity/Types.h

```diff
@@ -383,7 +386,15 @@ class FunctionType: public Type
 	virtual MemberList const& getMembers() const override;
 
 	Location const& getLocation() const { return m_location; }
-	std::string getCanonicalSignature(std::string const& _name) const;
+	/// @returns the canonical signature of this function type given the function name
+	/// If @a _name is not provided (empty string) then the @c m_declaration member of the
+	/// function type is used
+	std::string getCanonicalSignature(std::string const& _name = "") const;
+	Declaration const* getDeclaration() const { return m_declaration; }
+	bool isConstant() const { return m_isConstant; }
+	/// @return A shared pointer of an ASTString.
+	/// Can contain a nullptr in which case indicates absence of documentation
+	ASTPointer<ASTString> getDocumentation() const;
```

Comment:
It's not a part of the type. This is simply a kind of wrapper that will return the documentation if existing. It's only for convenience. If not for that you would have to get the declaration, see if it can be cast to `FunctionDefinition` and then attempt to get the Documentation.